lib/utilities/adapter-registry.js

```diff
@@ -36,5 +40,36 @@ module.exports = CoreObject.extend({
     if (addon.type === 'ember-deploy-addon') {
       this.adapters = merge(this.adapters, addon.adapters);
     }
+  },
+
+  _addAdaptersForTesting: function () {
```

Comment:
Support test shim for loading plugins not in the package.json. Downside with this approach is we bypass DAG which may cause some issues in the future with before/after hooks, etc.